Content.Server/GameObjects/Components/Mobs/Body/Limb.cs

```diff
@@ -0,0 +1,322 @@
+        {
+            State = LimbState.Missing;
+            foreach (var organ in Organs)
+            {
+                organ.State = OrganState.Dead;
+            }
+            foreach (var child in Children)
+            {
+                child.HandleDecapitation(childOrganDamage);
+            }
+            if (spawn)
+            {
+                SpawnPrototypeEntity();
+            }
+        }
+
+        private void Dispose()
+        {
+            Children = null;
+            Organs = null;
+        }
+
+        public void SpawnPrototypeEntity()
+        {
+            //TODO
+            if (!string.IsNullOrWhiteSpace(PrototypeEntity))
+            {
+                Owner.EntityManager.TrySpawnEntityAt(PrototypeEntity, Owner.Transform.GridPosition, out var entity);
+            }
+        }
+
+        public void HandleDamage(int damage) //TODO: test prob numbers, and add targetting!
+        {
+            if (State == LimbState.Missing)
+            {
+                return; //It doesn't exist, so it's unaffected by damage/heal
+            }
+            //if (_addTargets.Contains(target)) //handle the damage to additional snowflake targets (mouth, eyes, and whatever else you add)
+            var state = ChangeHealthValue(damage);
+            if(Organs.Count == 0)
+            {
+                return;
+            }
+            switch (state)
+            {
+                case LimbState.InjuredLightly:
+                    if(_seed.Prob(0.1f))
+                    {
+                        _seed.Pick(Organs).HandleDamage(damage);
+                    }
+                    break;
+                case LimbState.Injured:
+                    //Organs[0].HandleDamage(damage); //testing brain damage
+                    if (_seed.Prob(0.4f))
+                    {
+                        _seed.Pick(Organs).HandleDamage(damage);
+                    }
+                    break;
+                case LimbState.InjuredSeverely:
+                    _seed.Pick(Organs).HandleDamage(damage);
+                    break;
+                case LimbState.Missing:
+                    if (State != state)
+                    {
+                        HandleDecapitation(true);
+                    }
+                    break;
+            }
+            State = state;
+            Logger.DebugS("Limb", "Limb {0} received {1} damage!", Name, damage);
+
+        }
+
+        public Blood CirculateBlood(Blood blood)
+        {
+            blood.ChangeCurrentVolume(BloodChange);
+            return blood;
+        }
+
+        private LimbState ChangeHealthValue(int value)
+        {
+            CurrentHealth -= value;
+            if (CurrentHealth < 0)
+            {
+                CurrentHealth = 0;
+            }
+            if (CurrentHealth > MaxHealth)
+            {
+                CurrentHealth = MaxHealth;
+            }
+
+            switch ((float)CurrentHealth)
```

Comment:
You are also storing the limb state somewhere, but this code isn't explicitly setting it, meaning that it could be quite trivial for these values to become inconsistent.

You should make `State` a property in function of the current health instead of making it stored, and don't return it from this method.